Conversation List Highlighting

brianHarder opened the pull request:
<!--
Thanks for contributing to the project!
Please help us keep this project in good shape by going through this checklist.
Replace the empty checkboxes [ ] below with checked ones [X] as they are completed
Remember, you can preview this before saving it.
-->

### Contributor checklist:

- [X] My contribution is **not** related to translations.
- [X] My commits are in nice logical chunks with [good commit messages](http://chris.beams.io/posts/git-commit/)
- [X] My changes are [rebased](https://medium.com/free-code-camp/git-rebase-and-the-golden-rule-explained-70715eccc372) on the latest [`main`](https://github.com/signalapp/Signal-Desktop/tree/main) branch
- [X] A `pnpm run ready` run passes successfully ([more about tests here](https://github.com/signalapp/Signal-Desktop/blob/main/CONTRIBUTING.md#tests))
- [X] My changes are ready to be shipped to users

### Description

This PR ensures that highlighting in the chat list only occurs by mouse hover and keyboard navigation. Previously there was an issue where clicking and dragging on a chat would cause it to stay highlighted both throughout the drag and even after mouse release. To achieve this, the default mouse behavior has been prevented, and new CSS classes have been added to only highlight conversation lists via mouse hover or the tab key.

I manually tested this PR by performing mouse movements in my staging environment and ensuring that the highlighting was either applied or not applied as expected. I specifically simulated the click and drag several times to make sure the highlight was not lingering. I also added a test file - conversationListHighlight_test.ts - which simulates a hover movement to check for highlighting and a click-drag movement to check that the highlight does not stay. I did all my testing on my laptop, which runs macOS 10.15.7.

https://github.com/user-attachments/assets/1c4040c2-5071-4afc-b717-14c78c54dba1

trevor-signal commented on stylesheets/_modules.scss:
```diff
@@ -5306,6 +5305,23 @@ button.module-calling-participants-list__contact {
 
+      @include mixins.keyboard-mode {
+        &:focus:not(:disabled, &--disabled, &--is-selected) {
+          @include mixins.light-theme {
```
Why are we trying to set the colors to the opposite of the `hover` state above (I may just be not understanding).

Also, have you tested this in light mode? I'm not sure nested mixings work the way we always intend. We've been moving to the newer CSS syntax `light-dark()` which might work well here.

brianHarder commented:
@trevor-signal I believe I've addressed all outstanding feedback! Here are the changes:

1. Removed mock test
2. Removed change in BaseConversationListItem.tsx seeing that SCSS changes are sufficient
3. I originally had the colors for keyboard highlighting opposite because I was getting the wrong gradient in dark mode, and switching them fixed that but then introduced the wrong shade in light mode. I updated the code to use the light-dark() syntax as you suggested.


https://github.com/user-attachments/assets/f7dbb176-79e6-4f2e-aaab-4c3321480ed5
